### ISSUE TITLE
Basic Syntax Highlighting for common languages

### ISSUE THREAD

**Issue by achimmihca** [zarunbal/LogExpert#264]
# Feature
We often log XML, JSON, and other common data formats.
Would be nice if the log viewer has basic highlighting for these to easier find relevant info (e.g. which fields / values).

This should work together with mulitline log messages as well ( see #11 )

For an example how this could look like, see [LogViewPlus](https://www.logviewplus.com/) (scroll to "Syntax Highlighting" feature)

For performance reasons, highlighting should only be applied to visible parts of the log file.
Further, it needs to be multithreaded, i.e., don't block user actions while calculating the highlighting.

**Comment by Hirogen** [comment]
yeah that would be nice, probably something like notepad++, where you could choose the "language" you like to highlight.
